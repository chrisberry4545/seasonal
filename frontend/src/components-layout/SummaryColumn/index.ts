export * from './SummaryColumn';

export * from './SummaryColumn.interface';
