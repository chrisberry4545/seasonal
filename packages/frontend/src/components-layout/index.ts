export * from './ImageGridItem/ImageGridItem';

export * from './ImageGrid/ImageGrid';

export * from './SummaryColumn';

export * from './SummaryColumnList';

export * from './SeasonNameView/SeasonNameView';

export * from './Toast/Toast';
