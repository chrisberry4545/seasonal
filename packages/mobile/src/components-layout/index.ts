export * from './ConfirmButtonLayout/ConfirmButtonLayout';

export * from './DefaultPaddingContainer/DefaultPaddingContainer';

export * from './DetailsPageLayout/DetailsPageLayout';

export * from './FeedbackQuestionsLayout/FeedbackQuestionsLayout';

export * from './HeaderAndBackButton/HeaderAndBackButton';

export * from './ImageGrid/ImageGrid';

export * from './ImageGridItem/ImageGridItem';

export * from './MainContainer/MainContainer';

export * from './ModalLayout/ModalLayout';

export * from './NoResultsText/NoResultsText';

export * from './SeasonNameView/SeasonNameView';

export * from './SwitchableGridOrList/SwitchableGridOrList';

export * from './Toast/Toast';

export * from './TopLoadingSpinner/TopLoadingSpinner';
