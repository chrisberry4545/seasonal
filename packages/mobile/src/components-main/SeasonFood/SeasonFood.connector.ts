import {
  ISeasonFoodInputProps, ISeasonFoodDispatchProps
} from './SeasonFood.interface';

import { connect } from 'react-redux';
import {
  SeasonFood
} from './SeasonFood';
import {
  selectVisibleFoodData,
  foodItemClicked,
  selectIsCurrentSeasonFoodLoading,
  selectIsListViewShown,
  toggleListView
} from '@chrisb-dev/seasonal-shared-frontend-redux';
import { IState } from '../../store';
import { Dispatch } from 'redux';

const mapStateToProps = (
  state: IState
): ISeasonFoodInputProps => {
  return {
    food: selectVisibleFoodData(state),
    isListViewShown: selectIsListViewShown(state),
    isLoading: selectIsCurrentSeasonFoodLoading(state)
  };
};

const mapDispatchToProps = (
  dispatch: Dispatch
): ISeasonFoodDispatchProps => {
  return {
    onFoodClick: (foodItemId) => dispatch(foodItemClicked(foodItemId)),
    onToggleListView: () => dispatch(toggleListView())
  };
};

export const SeasonFoodConnector = connect(
  mapStateToProps,
  mapDispatchToProps
)(SeasonFood);
