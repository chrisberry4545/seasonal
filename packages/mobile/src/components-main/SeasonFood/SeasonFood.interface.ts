import {
  IFood
} from '@chrisb-dev/seasonal-shared-models';

export interface ISeasonFoodInputProps {
  isLoading: boolean;
  isListViewShown: boolean;
  food: IFood[] | undefined;
}

export interface ISeasonFoodDispatchProps {
  onFoodClick: (foodItemId: string) => void;
  onToggleListView: () => void;
}

export interface ISeasonFoodProps
  extends ISeasonFoodInputProps, ISeasonFoodDispatchProps {}
