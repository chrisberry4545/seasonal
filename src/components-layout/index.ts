export * from './ImageGrid/ImageGrid';

export * from './ImageGridItem/ImageGridItem';

export * from './MainContainer/MainContainer';

export * from './SeasonNameView/SeasonNameView';
