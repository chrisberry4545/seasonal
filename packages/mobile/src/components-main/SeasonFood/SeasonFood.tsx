import React, { FC } from 'react';

import {
  SeasonDetailsContentWrapperConnector
} from '../SeasonDetailsContentWrapper/SeasonDetailsContentWrapper.connector';
import { ISeasonFoodProps } from './SeasonFood.interface';

import {
  SwitchableGridOfList, TopLoadingSpinner
} from '../../components-layout';

export const SeasonFood: FC<ISeasonFoodProps> = ({
  isLoading,
  food,
  onFoodClick,
  isListViewShown,
  onToggleListView
}) => (
  <SeasonDetailsContentWrapperConnector>
    {
    !isLoading
      ? <SwitchableGridOfList
          isListViewShown={isListViewShown}
          onToggleListView={onToggleListView}
          data={food}
          onClick={onFoodClick} noResultsMessage='No results found' />
      : <TopLoadingSpinner />
    }
  </SeasonDetailsContentWrapperConnector>
);
