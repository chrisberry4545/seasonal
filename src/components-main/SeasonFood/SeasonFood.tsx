import React, { FC } from 'react';

import {
  SeasonDetailsContentWrapperConnector
} from './../SeasonDetailsContentWrapper/SeasonDetailsContentWrapper.connector';
import { ISeasonFoodProps } from './SeasonFood.interface';

import {
  ImageGrid, TopLoadingSpinner
} from '../../components-layout';

export const SeasonFood: FC<ISeasonFoodProps> = ({
  isLoading,
  food,
  onFoodClick
}) => (
  <SeasonDetailsContentWrapperConnector>
    {
    !isLoading
      ? <ImageGrid data={ food } onClick={onFoodClick} noResultsMessage='No results found' />
      : <TopLoadingSpinner />
    }
  </SeasonDetailsContentWrapperConnector>
);
