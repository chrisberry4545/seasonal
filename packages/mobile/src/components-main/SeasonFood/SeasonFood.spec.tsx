import React from 'react';
import { SeasonFood } from './SeasonFood';
import { shallow, ShallowWrapper } from 'enzyme';
import { TopLoadingSpinner, SwitchableGridOrList } from '../../components-layout';
import { IFood } from '@chrisb-dev/seasonal-shared-models';

jest.mock('../SeasonDetailsContentWrapper/SeasonDetailsContentWrapper.connector', () => ({
  SeasonDetailsContentWrapperConnector: () => 'SeasonDetailsContentWrapperConnector'
}));
jest.mock('../../components-layout', () => ({
  SwitchableGridOrList: () => 'SwitchableGridOrList',
  TopLoadingSpinner: () => 'TopLoadingSpinner'
}));

describe('<SeasonFood />', () => {
  let wrapper: ShallowWrapper;
  const food = [{}] as IFood[];
  let mockOnFoodClick: jest.Mock;
  let mockOnToggleListView: jest.Mock;
  beforeEach(() => {
    mockOnFoodClick = jest.fn();
    mockOnToggleListView = jest.fn();
  });

  describe('when it is not loading', () => {
    beforeEach(() =>
      wrapper = shallow(
        <SeasonFood
          isLoading={false}
          food={food}
          onFoodClick={mockOnFoodClick}
          isListViewShown={true}
          onToggleListView={mockOnToggleListView} />
      )
    );

    test('renders correctly', () =>
      expect(wrapper).toMatchSnapshot());

    test('can click on food items', () => {
      const onClick = wrapper.find(SwitchableGridOrList)
        .first().props().onClick;
      if (onClick) {
        onClick('foodId');
      }
      expect(mockOnFoodClick).toHaveBeenCalledWith('foodId');
    });

    test('can toggle list view', () => {
      const onToggleListView = wrapper.find(SwitchableGridOrList)
        .first().props().onToggleListView;
      if (onToggleListView) {
        onToggleListView();
      }
      expect(mockOnToggleListView).toHaveBeenCalled();
    });

  });

  describe('when it is loading', () => {
    beforeEach(() =>
      wrapper = shallow(
        <SeasonFood
          isLoading={true}
          food={food}
          onFoodClick={mockOnFoodClick}
          isListViewShown={true}
          onToggleListView={mockOnToggleListView} />
      )
    );

    test('shows a loading spinner', () =>
      expect(wrapper.find(TopLoadingSpinner).exists()).toBe(true));

  });

});
