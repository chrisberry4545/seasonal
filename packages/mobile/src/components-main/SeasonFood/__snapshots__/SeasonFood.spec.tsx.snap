// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeasonFood /> when it is not loading renders correctly 1`] = `
<SeasonDetailsContentWrapperConnector>
  <SwitchableGridOrList
    data={
      Array [
        Object {},
      ]
    }
    isListViewShown={true}
    noResultsMessage="No results found"
    onClick={[MockFunction]}
    onToggleListView={[MockFunction]}
  />
</SeasonDetailsContentWrapperConnector>
`;
